driver-mapping/src/main/java/com/datastax/driver/mapping/AnnotationParser.java

```diff
@@ -21,263 +21,170 @@
 import com.datastax.driver.mapping.annotations.*;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Strings;
-import com.google.common.collect.Maps;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.reflect.TypeToken;
 
+import java.beans.PropertyDescriptor;
 import java.lang.annotation.Annotation;
-import java.lang.reflect.Field;
-import java.lang.reflect.*;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
 import java.util.*;
 import java.util.concurrent.atomic.AtomicInteger;
 
 /**
- * Static metods that facilitates parsing class annotations into the corresponding {@link EntityMapper}.
+ * Static methods that facilitates parsing:
+ * - {@link #parseEntity(Class, MappingManager)}: entity classes into {@link EntityMapper} instances
+ * - {@link #parseUDT(Class, MappingManager)}: UDT classes into {@link MappedUDTCodec} instances.
+ * - {@link #parseAccessor(Class, MappingManager)}: Accessor interfaces into {@link AccessorMapper} instances.
  */
+@SuppressWarnings({"unchecked", "WeakerAccess"})
 class AnnotationParser {
 
-    private static final Comparator<Field> fieldComparator = new Comparator<Field>() {
+    @VisibleForTesting
+    static final Set<Class<? extends Annotation>> VALID_PROPERTY_ANNOTATIONS = ImmutableSet.of(
+            Column.class,
+            Computed.class,
+            ClusteringColumn.class,
+            Frozen.class,
+            FrozenKey.class,
+            FrozenValue.class,
+            PartitionKey.class,
+            Transient.class);
+
+    @VisibleForTesting
+    static final Set<Class<? extends Annotation>> VALID_FIELD_ANNOTATIONS = ImmutableSet.of(
+            Field.class,
+            Frozen.class,
+            FrozenKey.class,
+            FrozenValue.class,
+            Transient.class);
+
+    private static final Comparator<PropertyMapper> POSITION_COMPARATOR = new Comparator<PropertyMapper>() {
         @Override
-        public int compare(Field f1, Field f2) {
-            return position(f1) - position(f2);
+        public int compare(PropertyMapper o1, PropertyMapper o2) {
+            return o1.position - o2.position;
         }
     };
 
     private AnnotationParser() {
     }
 
-    public static <T> EntityMapper<T> parseEntity(Class<T> entityClass, EntityMapper.Factory factory, MappingManager mappingManager) {
+    static <T> EntityMapper<T> parseEntity(final Class<T> entityClass, MappingManager mappingManager) {
         Table table = AnnotationChecks.getTypeAnnotation(Table.class, entityClass);
 
-        String ksName = table.caseSensitiveKeyspace() ? table.keyspace() : table.keyspace().toLowerCase();
-        String tableName = table.caseSensitiveTable() ? table.name() : table.name().toLowerCase();
+        String ksName = table.caseSensitiveKeyspace() ? Metadata.quote(table.keyspace()) : table.keyspace().toLowerCase();
```

Comment:
> Was this an existing bug (That keyspaces / tables were not quoted)?

I believe so. In the current code, only UDT names were quoted here, the rest was quoted on demand later on.

> Had thought previously that you had to escape it in your annotation definition.

Well, there wasn't any test so far testing case sensitiveness, and nothing in the docs. From the code I infer that we do not expect users to quote identifiers themselves, i.e., a column `"Address"` would be declared as:

```java
@Column(name = "Address", caseSensitive = true)
```

And not:

```java
@Column(name = "\"Address\"", caseSensitive = true)
```

> Ah I see that EntityMapper used to do quoting on demand, where now it's done early, this change makes sense.

Same thing for `ColumnMapper`, which is the purpose of JAVA-564.